packages/transactions/src/signer.ts

```diff
@@ -19,6 +19,35 @@ export class TransactionSigner {
     this.originDone = false;
     this.checkOversign = true;
     this.checkOverlap = true;
+
+    // If multi-sig spending condition exists, iterate over
+    // auth fields and reconstruct sigHash
+    let spendingCondition = transaction.auth.spendingCondition;
+    if (spendingCondition && !isSingleSig(spendingCondition)) {
+
+      if (spendingCondition.fields.length >= spendingCondition.signaturesRequired) {
+        throw new Error('SpendingCondition has more signatures than are expected');
+      }
```

Comment:
I'd be careful with this check -- the fix from #1081 would also need to be applied here.